docs/BuildingAndTestingDXC.rst

```diff
@@ -93,6 +93,15 @@ You may substitute ``Ninja`` for ``Unix Makefiles`` to generate a makefile
 build. After generation completes you can run ``ninja`` or ``make`` as
 appropriate.
 
+To avoid some build failures such as missing `external/SPIRV-Headers`, make
+sure that you initialise the git submodules before running `ninja` or `make`:
+
+.. code-block:: sh
+
+  git submodule init
+  git submodule update
```

Comment:
This seems out of place. I would expect this to apply to all of the cmake commands above as well. However, it might be worth while mentioning the submodule somewhere. @llvm-beanz Where would you suggest we add the instruction to check out with code with the submodules?

Reply:
I indeed felt it was out of place. But it is in the place where **I** needed the information and that would have saved me quite a lot of time.

Please do feel free to put that information anywhere else.

Reply:
Separately I put together a PR cleaning up the documentation for building DXC in relation to #5762 which also added information about the submodules (see #6185).